### ISSUE TITLE
Why the train loss is negative?

### ISSUE THREAD

**Issue by Joevaen** [MIC-DKFZ/nnUNet#232]
I am a fresh of nnUnet, i want to know whether the negative loss is correct or not. And how to estimate the good result or bad? Very thanks!

> 2020-06-21 01:57:50.983468:
epoch:  162
2020-06-21 02:12:24.121007: train loss : -0.3924
2020-06-21 02:13:13.584377: validation loss: -0.4029
2020-06-21 02:13:13.585377: Average global foreground Dice: [0.66955129464171]
2020-06-21 02:13:13.585483: (interpret this as an estimate for the Dice of the different classes. This is not exact.)
2020-06-21 02:13:14.207725: lr: 0.00852
2020-06-21 02:13:14.207961: This epoch took 923.224429 s

2020-06-21 02:13:14.208023:
epoch:  163
2020-06-21 02:27:47.100379: train loss : -0.3893
2020-06-21 02:28:37.471998: validation loss: -0.4142
2020-06-21 02:28:37.472625: Average global foreground Dice: [0.7004898162719807]
2020-06-21 02:28:37.472711: (interpret this as an estimate for the Dice of the different classes. This is not exact.)
2020-06-21 02:28:38.059822: lr: 0.008511
2020-06-21 02:28:38.060060: This epoch took 923.851972 s

2020-06-21 02:28:38.060123:
epoch:  164
2020-06-21 02:43:10.191369: train loss : -0.3841
2020-06-21 02:44:01.761450: validation loss: -0.4186
2020-06-21 02:44:01.762121: Average global foreground Dice: [0.7061680657535285]
2020-06-21 02:44:01.762213: (interpret this as an estimate for the Dice of the different classes. This is not exact.)
2020-06-21 02:44:02.325695: lr: 0.008502
2020-06-21 02:44:02.325904: This epoch took 924.265718 s

2020-06-21 02:44:02.325975:
epoch:  165
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 8192.0
2020-06-21 02:58:32.567489: train loss : -0.3899
2020-06-21 02:59:24.175077: validation loss: -0.4195
2020-06-21 02:59:24.175920: Average global foreground Dice: [0.7040000833800485]
2020-06-21 02:59:24.176013: (interpret this as an estimate for the Dice of the different classes. This is not exact.)
2020-06-21 02:59:24.756553: lr: 0.008493
2020-06-21 02:59:24.756750: This epoch took 922.430723 s

**Comment by Joevaen** [comment]
This is Task_08 and I delete the cancer target part from the train-set.

**Comment by FabianIsensee** [comment]
the loss is the sum of CE and Dice loss. The best CE loss is 0, the best Dice loss is -1, so the overall best possible loss is -1.
I really don't understand why people expect  the loss to go to zero. That is pretty arbitrary :-) As long as it does down and does what it should the numerical value doesn't matter
Best,
Fabian

**Comment by FabianIsensee** [comment]
Your epoch time is suuuuper slow. You may want to investigate that. On a 2080 ti I would expect ~500s. On a V100 you should get ~200s or better

**Comment by Joevaen** [comment]
Very grateful for your patience and generosity.My training is excuted on RTX8000 with 48g memory.Maybe the time every epoch takes should be shorter? I will try to do an adjustment.Thanks.

**Comment by FabianIsensee** [comment]
RTX 8000 ist basically a 2080 ti with more memory, so it should be way faster. Maybe you are running into a CPU bottleneck

**Comment by Joevaen** [comment]
Thanks! Now I have a stupid question: if I do my own segement task based on aorta segementation in abdoment, are 40 ct series(about 300 slices in one) enough for training to get a not-bad result? If not, how many series should I prepare? And does the Task_08_HepaticVessel configuration also work here? If not, what should I do to rebuild this task.
Sorry to bother. For back, I will shared my experiments here to help bringing this wonderful frame up.Very thanks!
A little issue, is there a script used for generating the .json of the datasets in this frame? I am just verify if I need to code by myself.

**Comment by Joevaen** [comment]
> Thanks! Now I have a stupid question: if I do my own segement task based on aorta segementation in abdoment, are 40 ct series(about 300 slices in one) enough for training to get a not-bad result? If not, how many series should I prepare? And does the Task_08_HepaticVessel configuration also work here? If not, what should I do to rebuild this task.
> Sorry to bother. For back, I will shared my experiments here to help bringing this wonderful frame up.Very thanks!
> A little issue, is there a script used for generating the .json of the datasets in this frame? I am just verify if I need to code by myself.
